ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/text_embedding/TextEmbeddingModel.java

```diff
@@ -0,0 +1,239 @@
+        File modelZipFile = (File)params.get(MODEL_ZIP_FILE);
+        modelHelper = (ModelHelper)params.get(MODEL_HELPER);
+        if (modelZipFile == null) {
+            throw new IllegalArgumentException("model file is null");
+        }
+        if (modelHelper == null) {
+            throw new IllegalArgumentException("model helper is null");
+        }
+        modelId = model.getModelId();
+        if (modelId == null) {
+            throw new IllegalArgumentException("model id is null");
+        }
+        loadTextEmbeddingModel(
+                modelZipFile,
+                modelId,
+                model.getName(),
+                model.getAlgorithm(),
+                model.getVersion(),
+                model.getModelConfig(),
+                engine
+        );
+    }
+
+    @Override
+    public void close() {
+        if (modelHelper != null && modelId != null) {
+            modelHelper.deleteFileCache(modelId);
+            if (predictor != null) {
+                log.debug("close predictor for model {}", modelId);
+                predictor.close();
+                predictor = null;
+            }
+            if (model != null) {
+                log.debug("close model for model {}", modelId);
+                model.close();
+                model = null;
+            }
+        }
+    }
+
+    protected void loadTextEmbeddingModel(File modelZipFile, String modelId, String modelName, FunctionName functionName, Integer version,
+                                       MLModelConfig modelConfig,
+                                       String engine) {
+        try {
+            if (FunctionName.TEXT_EMBEDDING != functionName) {
+                throw new IllegalArgumentException("wrong function name");
+            }
+            if (!PYTORCH_ENGINE.equals(engine) && !ONNX_ENGINE.equals(engine)) {
+                throw new IllegalArgumentException("unsupported engine");
+            }
+            AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
+                ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+                try {
+                    System.setProperty("PYTORCH_PRECXX11", "true");
+                    System.setProperty("DJL_CACHE_DIR", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    System.setProperty("java.library.path", DJL_CACHE_PATH.toAbsolutePath().toString());
+                    Thread.currentThread().setContextClassLoader(ai.djl.Model.class.getClassLoader());
+                    Path modelPath = getModelCachePath(modelId, modelName, version);
+                    File pathFile = new File(modelPath.toUri());
+                    if (pathFile.exists()) {
+                        FileUtils.deleteDirectory(pathFile);
+                    }
+                    ZipUtils.unzip(new FileInputStream(modelZipFile), modelPath);
+                    boolean findModelFile = false;
```

Comment:
Why you need check file here?
You already know which engine to use. it doesn't hurt if the zip file contains multiple models, why need add this restriction?

Reply:
Will the engine throw error if have multiple model files? Or the engine will just load the first model file and ignore others?

Reply:
You specify the engine in the criteria, so it will only load the model with the engine you specified. All other files will be ignored.